lib/services/local/saml_idp_service_provider.go

```diff
@@ -256,7 +257,7 @@ func (s *SAMLIdPServiceProviderService) fetchAndSetEntityDescriptor(sp types.SAM
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return trace.Wrap(trace.ReadError(resp.StatusCode, nil))
+		return trace.NotFound("entity descriptor not found on the given endpoint")
```

Comment:
I'm not sure always returning "not found" here is appropriate here "semantically". It can be a different HTTP error, like internal server error or anything else, right? I would probably return "trace.BadParameter" here which we use as a generic error.

And also I would still include the response code in the error message, otherwise it's lost.

Reply:
Used `trace.BadParameter` in https://github.com/gravitational/teleport/pull/38825/commits/ee1b015ce0448c3684b544398e20d1aef21e0aa4. I avoided that initially as the error is not related user or system input.